packages/material-ui-styles/src/index.d.ts

```diff
@@ -82,6 +82,40 @@ declare module '@material-ui/styles/makeStyles' {
 
   // https://stackoverflow.com/a/49928360/3406963 without generic branch types
   type IsAny<T> = 0 extends (1 & T) ? true : false;
+  /**
+   * @internal
+   *
+   * check if a type is `{}`
+   *
+   * 1. false if the given type has any members
+   * 2. false if the type is `object` which is the only other type with no members
+   *  {} is a top type so e.g. `string extends {}` but not `string extends object`
+   */
+  export type IsEmptyInterface<T> = keyof T extends never
+    ? string extends T
+      ? true
+      : false
+    : false;
```

Comment:
That was my first approach too but that would evaluate to `true` for `object` or `Record<"a" | "b", string>`.  The first one isn't that problematic since it has no members either but the second one does.

Reply:
> That was my first approach too but that would evaluate to `true` for `object` or `Record<"a" | "b", string>`. The first one isn't that problematic since it has no members either but the second one does.

Strange, the second one shouldn't be true, and it [doesn't seem to be in the latest TS](http://www.typescriptlang.org/play/#src=type%20IsEmptyInterface%3CT%3E%20%3D%20%7B%7D%20extends%20T%20%3F%20true%20%3A%20false%3B%0A%0Aconst%20foo%3A%20IsEmptyInterface%3CRecord%3C%22a%22%20%7C%20%22b%22%2C%20string%3E%3E%20%3D%20false%3B%0A).

Reply:
Funny I did the same but somehow confused the evaluated type. Still [true for object though](http://www.typescriptlang.org/play/#src=type%20IsEmptyInterface%3CT%3E%20%3D%20%7B%7D%20extends%20T%20%3F%20true%20%3A%20false%3B%0A%0Aconst%20foo%3A%20IsEmptyInterface%3Cobject%3E%20%3D%20true%3B%0A). In that case I would prefer `keyof T extends never ? true : false` since it communicates intent better than `{} extends T` IMO.

Reply:
Right. Going to improve coverage and then stick to your approach most likely. `object` should be fine in most cases.

Reply:
@pelotom I extended coverage for `IsEmptyInterface` and tried improving readability. I tried your approach in https://github.com/eps1lon/material-ui/commit/60007f349d0c7aa07efe456291c46d38f34221cd but this doesn't work in its current form since it completely disables type checking for props.